### ISSUE TITLE
NPM dependency requests ^2.3.6 but seems to resolve 2.4.1?

### ISSUE THREAD

**Issue by AshWhitear** [enquirer/enquirer#437]
Hey All, I'm using AureliaJS which depends on the older version of enquirer. Today we've discovered our builds aren't working as expected, and after some digging it seems aurelia-js resolves the newer version of the library, despite requesting the old one. Are you guys able to help here - could this be related to the recent release?

```
...
"_from": "enquirer@^2.3.6",
"_id": "enquirer@2.4.1",
"_inBundle": false,
"_integrity": "sha512-..."
"_location": "/enquirer"
" _phantomChildren": {},
"_requested": {
    "type": "range"
    "registry": true,
    "paw" :
    "enquirer@^2.3.6",
    "name": "enquirer"
    "escapedName": "enquirer"
    "rawSpec": "^2.3.6",
    "saveSpec": null,
    "fetchSpec": "^2.3.6"
}, "_requiredBy": [
    "/aurelia-cli"
],
"_resolved": "https://registry.npmjs.org/enquirer/-/enquirer-2.4.1. tgz",
...
```

Thanks, Ash.

**Comment by jonschlinkert** [comment]
Hi @AshWhitear, have you tried deleting `node_modules` completely and reinstalling everything? If so, have you also tried deleting all lock files too?

**Comment by AshWhitear** [comment]
hi @jonschlinkert - thanks for the snappy response. I have indeed, this is happening on a pipeline agent which gets cleaned each time although I can replicate locally if I `npm install` then attempt to build with aurelia-cli. If I then follow that up installing the library directly with `npm install enquirer@2.3.6` it does collect the right version.

**Comment by jonschlinkert** [comment]
Thanks for clarifying. I just noticed the ^ in the version in package.json. That tells NPM to install the specified version or greater, up to, but not including, the next major version. Remove that to specify 2.3.6 only, then try reinstalling again. Sent from my iPhoneOn Sep 4, 2023, at 11:49 PM, AshWhitear ***@***.***> wrote:﻿
hi @jonschlinkert - thanks for the snappy response. I have indeed, this is happening on a pipeline agent which gets cleaned each time although I can replicate locally if I npm install then attempt to build with aurelia-cli. If I then follow that up installing the library directly with npm install ***@***.*** it does collect the right version.
